malduck/extractor/config_builder.py

```diff
@@ -0,0 +1,538 @@
+        usage: Optional[ConnUsageEnum] = None,
+    ) -> None:
+        """
+        SMTP connection used by malware
+
+        :param username: SMTP username
+        :param password: SMTP password
+        :param hostname: SMTP hostname
+        :param port: SMTP port
+        :param mail_to: E-mail addresses used by attacker to receive data
+        :param mail_from: E-mail address used by malware to send data
+        :param subject: E-mail subject
+        :param usage: Purpose of the connection
+        """
+        smtp = ExtractorModel.SMTP(
+            username=username,
+            password=password,
+            hostname=hostname,
+            port=port,
+            mail_to=mail_to or [],
+            mail_from=mail_from,
+            subject=subject,
+            usage=usage,
+        )
+        return self.push_config(dict(smtp=[smtp]))
+
+    def add_http(
+        self,
+        uri: Optional[str] = None,
+        protocol: Optional[str] = None,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        hostname: Optional[str] = None,
+        port: Optional[int] = None,
+        path: Optional[str] = None,
+        query: Optional[str] = None,
+        fragment: Optional[str] = None,
+        user_agent: Optional[str] = None,
+        method: Optional[str] = None,
+        headers: Optional[Dict[str, str]] = None,
+        max_size: Optional[int] = None,
+        usage: Optional[ConnUsageEnum] = None,
+    ) -> None:
+        """
+        HTTP connection used by malware
+
+        :param uri: |
+            HTTP(S) URI contained in configuration. If you're able to parse it, it's preferred to use
+            more detailed fields instead.
+
+        :param protocol: URI scheme (usually 'http' or 'https')
+        :param username: Username (e.g. for Basic auth)
+        :param password: Password
+        :param hostname: URI hostname
+        :param port: Port (if custom or provided explicitly in malware)
+        :param path: URI path
+        :param query: URI query (part that follows the '?')
+        :param fragment: URI fragment (part that follows the '#')
+        :param user_agent: Custom User-Agent header used for communication
+        :param method: HTTP method (GET/POST/...)
+        :param headers: Dictionary with additional headers required to contact C&C
+        :param max_size: Maximum accepted size by server
+        :param usage: Purpose of the connection
+        """
+        http = ExtractorModel.Http(
+            uri=uri,
+            protocol=protocol,
+            username=username,
+            password=password,
+            hostname=hostname,
+            port=port,
+            path=path,
+            query=query,
+            fragment=fragment,
+            user_agent=user_agent,
+            method=method,
+            headers=headers,
+            max_size=max_size,
+            usage=usage,
+        )
+        return self.push_config(dict(http=[http]))
```

Comment:
not compatible with ioc_extractor - should we add support to ioc_extractor for this, or ensure mwdb generates configs understood by ioc_extractor?